### PR TITLE
feat: `array-callback-return` supports `Array.prototype.toSorted`

### LINKED ISSUE #16842: Rule Change: `array-callback-return` should support Stage 4 `Array.prototype.toSorted``
### What rule do you want to change?

`array-callback-return`

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
[1, 2, 3].toSorted(
  (a, b) {
    if (cond1) return -1
    if (cond2) return 1
  }
);
```

### What does the rule currently do for this code?

Do nothing

### What will the rule do after it's changed?

Warns as for `Array.prototype.sort`.

### Participation

- [X] I am willing to submit a pull request to implement this change.

### Additional comments

Change array by copy proposal reaches to Stage 4
ref:  https://github.com/tc39/proposals/commit/ad4df8435f27f39eda26db3b940ae151980c8015

### PR DESCRIPTION
**Pull request by sosukesuzuki**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Fixes #16842 

Adds support for `Array.prototype.toSorted`

#### Is there anything you'd like reviewers to focus on?

N/A

### CODE REVIEW

File: docs/src/rules/array-callback-return.md
Comment (sosukesuzuki) [inline docs/src/rules/array-callback-return.md:38]: https://github.com/tc39/ecma262/pull/2997 hasn't been merged yet, so we cannot link to the ECMA-262.

**Review by snitin315 — APPROVED** [review]
LGTM, thanks for contributing.

I'll leave it open for others to review.

File: docs/src/rules/array-callback-return.md
Comment (mdjermanovic) [reply docs/src/rules/array-callback-return.md:38]: Maybe we should wait for https://github.com/tc39/ecma262/pull/2997 to be merged into the spec so that the new feature becomes official before merging this PR?

Comment (ljharb) [reply docs/src/rules/array-callback-return.md:38]: The feature is official the moment the committee grants it consensus; merging into the spec is ceremonial.

Comment (nzakas) [reply docs/src/rules/array-callback-return.md:38]: I don't see any reason to wait for the formal spec merge. We can always update the docs later.

**Review by nzakas — APPROVED** [review]
LGTM. Just want @mdjermanovic's okay before merging.

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

I agree with merging this now and updating the link later.

**Pull request merged**
